Automattic/wp-calypso
# Add-ons: Filter Storage Add On Options

This PR is based on https://github.com/Automattic/wp-calypso/pull/76754
Fixes 1721-gh-Automattic/martech

### Proposed Changes
Implement filtering for the available storage options based on the current max storage:

1. We want to offer 50 and 100 GB for now (this will likely change in the future)
2. A site should not have more than 200 GB of storage. If a user already has 200 GB of storage (e.g. provided by a plan), they should not have the option to add more storage.
3. Potentially, a user may be eligible for 50 GB, not 100 GB, depending on their current allowed storage.

### Testing
1. Checkout this branch
2. Run `yarn` & `yarn start`
3. Go to `http://calypso.localhost:3000/add-ons/YOUR_SITE_SLUG`
4. You should see the new card item for the storage add on:

![image](https://github.com/Automattic/wp-calypso/assets/20927667/a67ba66e-166c-4bd0-adb6-d084840182d9)

The values in the dropdown should be dynamic depending on the current storage usage for a site.
